include/Dialect/LWE/IR/LWEAttributes.td

```diff
@@ -55,10 +67,165 @@ def LWE_BitFieldEncoding: LWE_EncodingAttr<"BitFieldEncoding", "bit_field_encodi
+    #coeff_encoding = #lwe.poly_coefficient_encoding<cleartext_start=15, cleartext_bitwidth=4>
+
+    %poly1 = poly.from_tensor %coeffs1 : tensor<10xi16> -> !poly.poly<#ring>
+    %poly2 = poly.from_tensor %coeffs2 : tensor<10xi16> -> !poly.poly<#ring>
+    %rlwe_ciphertext = tensor.from_elements %poly1, %poly2 : tensor<2x!poly.poly<#ring>, #coeff_encoding>
+    ```
+  }];
 }
+
+def RLWE_PolyEvaluationEncoding
+  : LWE_EncodingAttrWithScalingFactor<"PolyEvaluationEncoding", "poly_evaluation_encoding"> {
+  let summary = "An attribute describing encoded RLWE plaintexts via evaluations at fixed points.";
+  let description = [{
+    A "evaluation encoding" of a list of integers $(v_1, \dots, v_n)$ asserts
+    that $f(\x_1) = v_1, \dots, f(x_n) = v_n$ for some implicit, but fixed and
+    distinct choice of inputs $x_i$. The encoded values are also scaled by a
+    scale factor, having the same semantics as `bit_field_encoding`, but
+    applied entry-wise (to either the coefficient or evaluation representation).
+
+    This attribute can be used in multiple ways:
+
+    - On a `poly.poly`, it asserts that the polynomial has been transformed
+      from an evaluation tensor.
+    - On a tensor of `poly.poly`, it asserts that the tensor is an RLWE
+      ciphertext for some RLWE scheme that supports the evaluation encoding.
+
+    A typical workflow for the BFV/BGV schemes using this encoding would be
+    to apply a INTT operation to the input list of cleartexts to convert from
+    evaluation form to coefficient form, then encrypt the resulting polynomial
+    in coefficient form, then apply NTT back to the evaluation form for faster
+    multiplication of ciphertexts.
+
+    The points chosen are fixed to be the powers of a primitive root of unity
+    of the coefficient ring of the plaintext space, which allows one to use
+    NTT/INTT to tansform quickly between the coefficient and evaluation forms.
+
+    Example:
+
+    ```
+    #generator = #poly.polynomial<1 + x**1024>
+    // note that the cmod should be chosen so as to ensure a primitive root of
+    // unity exists in the multiplicative group (Z / cmod Z)^*
+    #ring = #poly.ring<cmod=65536, ideal=#generator>
+    #lwe_encoding = #lwe.poly_evaluation_encoding<cleartext_start=30, cleartext_bitwidth=3>
+
+    %evals = arith.constant <[1, 2, 4, 5]> : tensor<4xi16>
+    // Note no `intt` operation exists in poly yet.
+    %poly1 = poly.intt %evals : tensor<4xi16> -> !poly.poly<#ring, #eval_encoding>
+    %poly2 = poly.intt %evals : tensor<4xi16> -> !poly.poly<#ring, #eval_encoding>
+    %rlwe_ciphertext = tensor.from_elements %poly1, %poly2 : tensor<2x!poly.poly<#ring, #eval_encoding>>
+    ```
+  }];
+}
+
+// TODO: does it make sense to use the bit-field parameters here? It may
+// not work sensibly because the rounding happens after scaling-post-INTT,
+// though you can still use the parameters in this attribute to infer the
+// right power of two to use.
```

Comment:
I'm not sure I totally understand why it's bad to use bit-field parameters here - just because it's defining the `Delta` scaling that occurs after doesn't seem to preclude its use here (I think the semantics of bit-field encoding still applies here, and it's part of "inverse_canonical_embedding_encoding" that it occurs after the inverse canonical embedding operation). Maybe I'm misunderstanding this comment.

Reply:
I think the restriction to powers of 2 may not make sense here, since the scaling factor is not happening via bit shifting at this stage. I believe the scaling happens before rounding to integers. If there is a need for non-power-of-two scaling, then we would want different parameters. I'll file an issue